src/reactviews/pages/QueryResult/table/plugins/autoColumnSize.plugin.ts

```diff
@@ -0,0 +1,383 @@
+                columnDefs,
+                colIndices,
+            );
+
+            for (let i = 0; i < columnDefs.length; i++) {
+                let colIndex: number = colIndices[i];
+                let column: Slick.Column<T> = allColumns[colIndex];
+                let autoSizeWidth: number =
+                    Math.max(headerWidths[i], maxColumnTextWidths[i]) + 1;
+                if (autoSizeWidth !== column.width) {
+                    allColumns[colIndex].width = autoSizeWidth;
+                    change = true;
+                }
+            }
+
+            if (change) {
+                this.onPostEventHandler.unsubscribeAll();
+                this._grid.setColumns(allColumns);
+                this._grid.onColumnsResized.notify();
+            }
+        }
+    }
+
+    private handleDoubleClick(e: JQuery.TriggeredEvent<HTMLElement, unknown>) {
+        let headerEl = jQuery(e.currentTarget).closest(".slick-header-column");
+        let columnDef = headerEl.data("column");
+
+        if (!columnDef || !columnDef.resizable) {
+            return;
+        }
+
+        e.preventDefault();
+        e.stopPropagation();
+
+        this.reSizeColumn(headerEl, columnDef);
+    }
+
+    private reSizeColumn(headerEl: JQuery, columnDef: Slick.Column<T>) {
+        let headerWidth = this.getElementWidths([headerEl[0]])[0];
+        let colIndex = this._grid.getColumnIndex(columnDef.id!);
+        let origCols = this._grid.getColumns();
+        let allColumns = deepClone(origCols);
+        allColumns.forEach((col, index) => {
+            col.formatter = origCols[index].formatter;
+            col.asyncPostRender = origCols[index].asyncPostRender;
+        });
+        let column = allColumns[colIndex];
+
+        let autoSizeWidth =
+            Math.max(
+                headerWidth,
+                this.getMaxColumnTextWidth(columnDef, colIndex),
+            ) + 1;
+
+        // Only resize if the current width is smaller than the new width.
+        if (autoSizeWidth > column.width!) {
+            allColumns[colIndex].width = autoSizeWidth;
+            this._grid.setColumns(allColumns);
+            this._grid.onColumnsResized.notify();
+        }
+    }
+
+    /**
+     * For each column, find the max width of the texts in the first MAX_ROW_TO_SCAN rows.
+     * @param columnDefs Column definitions of all columns that need to be resized
+     * @param colIndices Column indices of all columns that need to be resized
+     * @returns An array of the max widths of each column
+     */
+    private getMaxColumnTextWidths(
+        columnDefs: Slick.Column<T>[],
+        colIndices: number[],
+    ): number[] {
+        let data = this._grid.getData() as Slick.DataProvider<T>;
+        let dataLength = data.getLength();
+        let viewPort = this._grid.getViewport();
+        let start = Math.max(0, viewPort.top);
+        let end = Math.min(dataLength, viewPort.bottom);
+        if (end < 50 && dataLength >= 50) {
```

Comment:
add a comment explaining what's going on here, something like
`// limit column width calculation to 50 rows`